### ISSUE TITLE
`typesafe_array_init` rule reports violations as though they were of the `array_init` rule.

### ISSUE THREAD

**Issue by mildm8nnered** [realm/SwiftLint#5709]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The `TypesafeArrayInitRule` outsources some of its functionality to the `ArrayInitRule`. The violations are reported as being of `array_init` rather than `typesafe_array_init`.

##### Complete output when running SwiftLint, including the stack trace and command used

`typesafe_array_init` is an analyzer rule, so it's hard to run the analyzer on it in isolation, but the following unit test illustrates the problem:

```
final class TypesafeArrayInitRuleTests: SwiftLintTestCase {
    func testViolationRuleIdentifier() {
        let baseDescription = TypesafeArrayInitRule.description
        guard let triggeringExample = baseDescription.triggeringExamples.first else {
            XCTFail("No triggering examples found")
            return
        }
        guard let config = makeConfig(nil, baseDescription.identifier) else {
            XCTFail("Failed to create configuration")
            return
        }
        let violations = SwiftLintFrameworkTests.violations(triggeringExample, config: config, requiresFileOnDisk: true)
        XCTAssertGreaterThanOrEqual(violations.count, 1)
        XCTAssertEqual(violations.first?.ruleIdentifier, baseDescription.identifier)
    }
}
```

This will fail the final check, as the types of the violations will not be `typesafe_array_init`

**Comment by mildm8nnered** [comment]
Resolved by #5710
